libtor/src/lib.rs

```diff
@@ -20,6 +20,7 @@
 #[macro_use]
 extern crate libtor_derive;
 extern crate log as log_crate;
+extern crate rand;
```

Comment:
I think you are missing `extern crate sha1` here?

Reply:
OK, I added.